core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/dml.scala

```diff
@@ -38,13 +38,52 @@ case class UpdateTable(
   override def output: Seq[Attribute] = target.output
 }
 
-case class MergeTables(
+/**
+ * The logical plan of the MERGE INTO command, aligned with SparkSQL
+ */
+case class MergeIntoTable(
     target: LogicalPlan,
-    source: Option[LogicalPlan],
-    conditions: Option[Expression],
+    source: LogicalPlan,
+    mergeCondition: Expression,
+    matchedActions: Seq[MergeAction],
+    notMatchedActions: Seq[MergeAction],
+    notMatchedBySourceActions: Seq[MergeAction],
     outputRelation: Option[LogicalPlan],
     options: Option[Expression])
     extends Modification {
-  override def children: Seq[LogicalPlan] = Seq(target, source.getOrElse(NoopNode), outputRelation.getOrElse(NoopNode))
+
+  override def children: Seq[LogicalPlan] = {
+    val notMatchedValues = notMatchedActions.collect { case insertAction: InsertAction =>
+      insertAction.values
+    }
+    Seq(target, source, outputRelation.getOrElse(NoopNode)) ++ notMatchedValues
+  }
   override def output: Seq[Attribute] = target.output
 }
+
+sealed abstract class MergeAction extends Expression {
+  def condition: Option[Expression]
+  override def dataType: DataType = UnresolvedType
+  override def children: Seq[Expression] = condition.toSeq
+}
+
+case class DeleteAction(condition: Option[Expression]) extends MergeAction
+
+case class UpdateAction(condition: Option[Expression], assignments: Seq[Expression]) extends MergeAction {
+  override def children: Seq[Expression] = condition.toSeq ++ assignments
+}
+
+case class UpdateStarAction(condition: Option[Expression]) extends MergeAction {
+  override def children: Seq[Expression] = condition.toSeq
+}
+
+// TODO: As per Insert Above, the columns and values should perhaps become assignments to match Spark,
+// but TSql allows INSERT from derived rows, so this may be a better representation.
+case class InsertAction(condition: Option[Expression], columns: Option[Seq[Id]], values: LogicalPlan)
```

Comment:
has to be
```suggestion
case class InsertAction(condition: Option[Expression], assignments: Seq[Assign])
```
make a TODO on `Assign` that is has to be renamed to `Assignment`, as this is `x = y` in `UPDATE a SET x = y`

Reply:
Done